dbt/node_runners.py

```diff
@@ -291,42 +291,50 @@ def run_hooks(cls, project, adapter, flat_graph, hook_type):
 
         nodes = flat_graph.get('nodes', {}).values()
         hooks = get_nodes_by_tags(nodes, {hook_type}, NodeType.Operation)
-
-        # This will clear out an open transaction if there is one.
-        # on-run-* hooks should run outside of a transaction. This happens b/c
-        # psycopg2 automatically begins a transaction when a connection is
-        # created. TODO : Move transaction logic out of here, and implement
-        # a for-loop over these sql statements in jinja-land. Also, consider
-        # configuring psycopg2 (and other adapters?) to ensure that a
-        # transaction is only created if dbt initiates it.
-        conn_name = adapter.clear_transaction(profile)
+        adapter.clear_transaction(profile)
```

Comment:
@cmcarthur this will use the `master` connection, right? I think this can still cause issues with the `on-run-end` hook after a long dbt run. Do we need to clear this transaction here?

Reply:
I guess not? good call